### PR TITLE
Fix `FindCFramePercent`.

### PR DESCRIPTION
**Pull request by Jarod42**
Add extra checks for construction

### CODE REVIEW

File: src/action/action_built.cpp [hunk 1]
```diff
@@ -325,8 +325,8 @@ static std::size_t FindCFramePercent(const std::vector<CConstructionFrame> &cfra
 {
 	const auto it =
 		ranges::find_if(cframes, [&](const auto &frame) { return percent < frame.Percent; });
```
Comment (zzam) [inline src/action/action_built.cpp:327]: I did not notice before that there is a (backported) ranges-header in here :-)
Nice

Comment (zzam) [inline src/action/action_built.cpp:327]: I played with unit-testing this spot, but it got quite large. Not sure if we should add it.

Comment (Jarod42) [reply src/action/action_built.cpp:327]: I didn't look a lot for UTs here :/
Responsability split between Construction and action...

Comment (zzam) [reply src/action/action_built.cpp:327]: I will clean the test up and create a PR (maybe today).

**Pull request merged**